aea/cli/remove.py

```diff
@@ -396,8 +408,15 @@ def _remove_from_config(self) -> None:
             "Removing the {} from {}".format(self.item_type, DEFAULT_AEA_CONFIG_FILE)
         )
         self.agent_items.remove(current_item)
+        self.agent_config.component_configurations.pop(
+            ComponentId(self.item_type, current_item), None
+        )
+        self._dump_agent_config()
```

Comment:
why are we doing this here?...should it not be enough to have the call `remove_unused_component_configurations` at the end?

Reply:
thanks a lot, this outdated part of code made a huge trick. the logic was significantly updated

Reply:
tests were improved